File: lms-frontend/src/services/enrollmentService.js
import api from "./api";

const enrollmentService = {
  enrollStudent: async (courseId, studentId) => {
    const response = await api.post(`/api/courses/${courseId}/enroll`, {
      studentId,
    });
    return response.data;
  },

  getEnrolledCourses: async (studentId) => {
    const response = await api.get(`/api/students/${studentId}/courses`);
    return response.data; // List of CourseResponseDTO
  },
};

export default enrollmentService;

File: course-service/src/main/java/com/lms/courseservice/service/EnrollmentService.java
package com.lms.courseservice.service;

import com.lms.courseservice.dto.response.CourseResponseDTO;
import java.util.List;

public interface EnrollmentService {
    void enrollStudent(Long courseId, Long studentId);
    List<CourseResponseDTO> getEnrolledCourses(Long studentId);
}
